Extensions/TweenBehavior/tweenruntimebehavior.js

```diff
@@ -40,65 +41,60 @@ gdjs.TweenRuntimeBehavior.ObjectPositionTweenInstance = function (instance, hasF
   this.y = y;
 }
 
-gdjs.TweenRuntimeBehavior._addObjectVariableTween = (owner, identifier, instance, variable, currentValue) => {
-  gdjs.TweenRuntimeBehavior._tweenList[owner][identifier] = new gdjs.TweenRuntimeBehavior.ObjectVariableTweenInstance(instance, false, 0, variable, currentValue);
+gdjs.TweenRuntimeBehavior.prototype._addObjectVariableTween = (identifier, instance, variable, currentValue) => {
+  this._tweens[identifier] = new gdjs.TweenRuntimeBehavior.ObjectVariableTweenInstance(instance, false, 0, variable, currentValue);
 };
 
-gdjs.TweenRuntimeBehavior._addObjectPositionTween = (owner, identifier, instance, x, y) => {
-  gdjs.TweenRuntimeBehavior._tweenList[owner] = { identifier };
-  gdjs.TweenRuntimeBehavior._tweenList[owner][identifier] = new gdjs.TweenRuntimeBehavior.ObjectPositionTweenInstance(instance, false, 0, x, y);
+gdjs.TweenRuntimeBehavior.prototype._addObjectPositionTween = (identifier, instance, x, y) => {
+  this._tweens[identifier] = new gdjs.TweenRuntimeBehavior.ObjectPositionTweenInstance(instance, false, 0, x, y);
 };
 
-gdjs.TweenRuntimeBehavior._getTween = (owner, identifier) => {
-  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier];
+gdjs.TweenRuntimeBehavior.prototype._getTween = (identifier) => {
+  return this._tweens[identifier];
 };
 
-gdjs.TweenRuntimeBehavior._tweenExists = (owner, identifier) => {
-  if (owner in gdjs.TweenRuntimeBehavior._tweenList) {
-    if (identifier in gdjs.TweenRuntimeBehavior._tweenList[owner]) {
-      return true
-    }
+gdjs.TweenRuntimeBehavior.prototype._tweenExists = (identifier) => {
+  if (identifier in this._tweens) {
```

Comment:
I have a new issue. `this._tweens` is always `undefined` when I try to access it from within a function. If I access it from within the `addObjectPositionTween` function it works correctly.
Do I need to access the `_tweens` map differently from within another function?

Reply:
* `addObjectPositionTween` is defined on `gdjs.TweenRuntimeBehavior.prototype`
* `_tweenExists` is defined on `gdjs.TweenRuntimeBehavior.prototype` too. 
* You're properly calling `this._tweenExists`. So they should work the same.

What to do then? Fire up the Chrome/Electron debugger!
Add a breakpoint. Inspect `this`.
Add a `console.log(this)` in addObjectPositionTween and then in _tweenExists. Check if both seems equal.
Add a `console.log(this._tweens)` in both. See what is output to confirm that both are defined or not. If both are undefined, the problem is somewhere else.

Basically, open the console/debugger and test your assumption until you find the cause. If you don't find anything, post here every console.log/output/test that you've made so that we can help.

Reply:
Nothing related, but `in` is a bit special and goes through the prototype chain to see if inherited properties can be found. Leading to weird (but totally valid) things like `'toString' in {}; // returns true` (yeah I know ^^').
Solution is simple though! Simply bracket accessor notation: `this._tweens[identifier]` will return undefined if identifier is not defined on the object.
You can then "cast" this into a boolean by prepending a !! (the first ! will double negate the expression and return a boolean). undefined is roughly equal to false in JS, and any object/function/array is roughly equal to true. So at the end 

```
gdjs.TweenRuntimeBehavior.prototype._tweenExists = (identifier) => {
  return !!this._tweens[identifier];
}
```

Finally word, I see that you're using features like arrow functions and const/let. Will it's the way to go in the IDE, in the game engine we're potentially running in older browsers without supporting for it. Hence we might want to avoid them for compatibility (i.e: resort to using `function` keyword and `var`).

Reply:
@Wend1go Found the issue while writing the last comment.

You've been using an arrow functions. Arrow functions in JS are not binding `this` to the currently called function (contrary to functions). While this is super extra useful while writing modern JS with React (like in the IDE), it's also here not working because we're not in a class, we're just declaring manually a class using the "old way" of doing it in JS (by using `prototype` and `function`) (I say old, but under the hood these are equivalents).

So to fix this:
* Use `function` everywhere when declaring "methods" in the game engine extension files.

We might switch later to a more modern JavaScript syntax with class and arrow functions, but this is when all major browser will support this.

Reply:
@4ian this is gold for documentation. Really valuable feedback too